app/ts/background/background.ts

```diff
@@ -160,22 +160,18 @@ async function handleSimulationMode(
 	request: InterceptedRequest,
 	settings: Settings
 ): Promise<HandleSimulationModeReturnValue> {
-	let parsedRequest // separate request parsing and request handling. If there's a parse error, throw that to API user
-	try {
-		parsedRequest = EthereumJsonRpcRequest.parse(request.options)
-	} catch (error) {
+	const maybeParsedRequest = EthereumJsonRpcRequest.safeParse(request.options)
+	if (maybeParsedRequest.success === false) {
 		console.log(request)
```

Comment:
This log needed?

Reply:
yeah, I want to log the errors, they are quite handy when dapp sends weird requests at times